lib/resty/openssl/x509/extension.lua

```diff
@@ -225,7 +225,11 @@ function _M.to_data(extension, nid)
     local got = ffi_cast("GENERAL_NAMES*", void_ptr)
     local lib = require("resty.openssl.x509.altname")
     -- the internal ptr is returned, ie we need to copy it
-    return lib.dup(got)
+    local res, err = lib.dup(got)
```

Comment:
The purpose of `dup` is to make a copy that independent of the old instance, so we don't need to keep track of the old instance here. How do you think?

Reply:
I agree with you. But the tail call (return lib.dup) will make the void_ptr as a dead GC object. And it will be GC collected before entering into the lib.dup().

You can set the following environment and run the test.
There are still some issues I don't fix yet.

```shell
export TEST_NGINX_INIT_BY_LUA="debug.sethook(function () collectgarbage() end, 'l') jit.off()"
export TEST_NGINX_USE_VALGRIND=1
```

Reply:
I see, we can keep the
```
local res, err = lib.dup(got)
return res, err
```
pattern and remove the reference of void_ptr